src/ioc.test.ts

```diff
@@ -195,6 +195,11 @@ describe("ioc tests", () => {
     assert.strictEqual(iocT9.get("firstDep"), defaultDependencyOne);
     assert.strictEqual(iocT9.get("secondDep"), defaultDependencyTwo);
 
+    // SetDeps does not touch any dependency whose name was not passed in the call
+    assert.strictEqual((iocT9.setDeps(overrideDependencyOneObject, true))["firstDep"], iocT9.get("firstDep"));
+    assert.strictEqual(iocT9.get("secondDep"), defaultDependencyTwo);
+    iocT9.reset();
+
```

Comment:
I like your structure just above where you have:
- `// reset one dependency`
- `// reset all dependencies`.

We could do a similar thing here where we 
- `// override one default`, which is this new bit
- `// override all defaults` which is currently lines 181-183.

I'll leave it up to you, just a thought.